loader/loader.go

```diff
@@ -170,16 +171,80 @@ func (l *Loader) Load(patterns ...string) ([]*GunkPackage, error) {
 			// A Go package that isn't a Gunk package - skip it.
 			continue
 		}
+		if err := validatePackage(pkg); err != nil {
+			return nil, err
+		}
 		if l.cache == nil {
 			l.cache = make(map[string]*GunkPackage)
 		}
 		l.cache[pkg.PkgPath] = pkg
 		pkgs = append(pkgs, pkg)
 	}
-
 	return pkgs, nil
 }
 
+// validatePackage will validate that a given GunkPackage. It should
+// check sanity checks and common error cases that are shared with
+// the format and generate packages.
+func validatePackage(gunkPackage *GunkPackage) (formatErr error) {
+	// Use custom panic values to report errors from the inspect func,
+	// since that's the easiest way to immediately halt the process and
+	// return the error.
+	type inspectError struct{ err error }
+	defer func() {
+		if r := recover(); r != nil {
+			if ierr, ok := r.(inspectError); ok {
+				formatErr = ierr.err
+			} else {
+				panic(r)
+			}
+		}
+	}()
+	for _, file := range gunkPackage.GunkSyntax {
+		ast.Inspect(file, func(node ast.Node) bool {
+			switch node.(type) {
+			case *ast.StructType:
```

Comment:
use an if-not-ok-early-return and unindent  the rest of the code

Reply:
Done.